code/datums/chatmessage.dm

```diff
@@ -68,7 +74,7 @@
 /datum/chatmessage/proc/generate_image(text, atom/target, mob/owner, list/extra_classes, lifespan)
 	// Register client who owns this message
 	owned_by = owner.client
-	RegisterSignal(owned_by, COMSIG_PARENT_QDELETING, .proc/qdel, src)
+	RegisterSignal(owned_by, COMSIG_PARENT_QDELETING, .proc/parent_qdel)
```

Comment:
```suggestion
	RegisterSignal(owned_by, COMSIG_PARENT_QDELETING, .proc/pn_parent_qdel)
```
Last suggestion before merge. Make this clear, else it implies the proc qdels the parent.